cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/ConsumerSupplier.java

```diff
@@ -95,10 +98,17 @@ void open(int groupSize) {
         consumerConfig = queueConsumer.getConfig();
         consumer = queueConsumer;
       } else {
-        try {
-          usageRegistry.register(program, queueName.toStreamId());
-        } catch (Exception e) {
-          LOG.warn("Failed to register usage of {} -> {}", program, queueName.toStreamId(), e);
+        for (Id owner : owners) {
+          try {
+            // TODO: redundant code
+            if (owner instanceof Id.Program) {
```

Comment:
This is a repeated pattern 3-4 times. can we do a comparison with ``owner.getIdType()``? Also, I see that ``UsageRegistry.register`` also does this check right... can we just have the check there, and send whatever object you get from the callers?